src/spikeinterface/sorters/external/kilosort4.py

```diff
@@ -186,10 +224,22 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
         recording = cls.load_recording_from_folder(sorter_output_folder.parent, with_warnings=False)
         probe = load_probe(probe_path=probe_filename)
         probe_name = ""
-        filename = ""
 
-        # this internally concatenates the recording
-        file_object = RecordingExtractorAsArray(recording_extractor=recording)
+        if params["use_binary_file"]:
```

Comment:
Actually, if the recording is binary I would always do this path. It's faster.

@zm711 @samuelgarcia do you agree?

Reply:
That makes sense to me. KS4 likes the binary so if that's an option I think feeding it that is better.

Reply:
done! the only *caveat* is that if one has a binary recording, but wants to force the `RecordingExtractorAsArray` mode, the only way is to modify the recording, e.g.: `recording_non_bin = spre.scale(recording_bin)`

Reply:
done and added a test that checks that results are the same: https://github.com/chrishalcrow/spikeinterface/blob/only-allow-above-ks-4-16/.github/scripts/test_kilosort4_ci.py#L401